pkg/services/hivelocity/device/device.go

```diff
@@ -302,135 +218,48 @@ func (s *Service) deviceExists(ctx context.Context, deviceID int32) (bool, error
 	return false, nil
 }
 
-func chooseAvailableFromList(ctx context.Context, devices []*hv.BareMetalDevice, deviceType infrav1.HivelocityDeviceType, clusterName string) (*hv.BareMetalDevice, error) {
-	log := ctrl.LoggerFrom(ctx)
+func chooseAvailableFromList(devices []*hv.BareMetalDevice, deviceType infrav1.HivelocityDeviceType, clusterName, machineName string) (*hv.BareMetalDevice, error) {
 	for _, device := range devices {
 		// Ignore if associated already
-		if isAssociated(device) {
+		machineTag, err := hvtag.MachineTagFromList(device.Tags)
+		if err != nil && !errors.Is(err, hvtag.ErrDeviceTagNotFound) {
+			// unexpected error - continue
+			continue
+		}
+		if machineTag.Value == machineName {
+			// associated to this machine - return
+			return device, nil
+		}
+		if machineTag.Value != "" {
+			// associated to other machine - continue
 			continue
 		}
 
-		dt, err := getDeviceType(device)
+		deviceTypeTag, err := hvtag.DeviceTypeTagFromList(device.Tags)
 		if err != nil {
-			if errors.Is(err, errNoMatchingTagFound) {
-				continue
-			}
-			// don't return on err, otherwise a single broken device would block this method
-			log.Error(err, "[chooseAvailableFromList] getDeviceType() failed")
+			// TODO: What if no device type has been set? Is the device available or not?
 			continue
 		}
 
 		// Ignore if has wrong device type
-		if dt != string(deviceType) {
+		if deviceTypeTag.Value != "" && deviceTypeTag.Value != string(deviceType) {
```

Comment:
I think `deviceTypeTag.Value != "" &&` is not needed. 

I guess `if deviceTypeTag.Value != string(deviceType) {...` is enough.

Reply:
done